packages/client/src/detect-proxy.js

```diff
@@ -0,0 +1,71 @@
+import { exec } from 'node:child_process';
+import { promisify } from 'node:util';
+import logger from '@percy/logger';
+
+export default class DetectProxy {
+  constructor() {
+    this.execPromise = promisify(exec);
+    this.platform = process.platform;
+    // There are sock proxies as well which we don't need
+    this.filter = ['HTTP', 'HTTPS'];
+  }
+
+  async getSystemProxy() {
+    if (this.platform === 'darwin') {
+      return this.getProxyFromMac();
+    } else if (this.platform === 'win32') {
+      return this.getProxyFromWindows();
+    } else if (this.platform !== 'linux') {
+      logger('client:detect-proxy').debug(`Not able to auto detect system proxy for ${this.platform} platform`);
+    }
+  }
+
+  async getProxyFromMac() {
```

Comment:
Can you add example of string which you are splitting to understand the structure

Reply:
Added for both windows and mac